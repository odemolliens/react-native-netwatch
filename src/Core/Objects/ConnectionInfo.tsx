import { ILog, LogType } from '../../types';
import { NetInfoState } from '@react-native-community/netinfo/src/internal/types';

export class ConnectionInfo implements ILog {
  _id: number = -1;
  type: LogType = 'CONNECT';
  startTime: number = Date.now();
  connection: NetInfoState | null = null;

  constructor(attributes?: any) {
    Object.assign(this, attributes);
  }
}

export default ConnectionInfo;
