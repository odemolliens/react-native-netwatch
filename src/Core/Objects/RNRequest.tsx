import { ILog, LogType } from '../../types';
// @ts-ignore
// BlobFileReader is needed to read the content of the body response (it's type can be blob)
import BlobFileReader from 'react-native/Libraries/Blob/FileReader';

const fromEntries = (arr: any[]) =>
  arr.reduce((acc, [k, v]) => {
    acc[k] = v;
    return acc;
  }, {});

export const stringifyData = (data: any) => {
  try {
    if (data?._parts?.length) {
      return JSON.stringify(fromEntries(data?._parts), null, 2);
    }
    return JSON.stringify(JSON.parse(data), null, 2);
  } catch (e) {
    return `${data}`;
  }
};

export const getRequestBody = (dataSent: any): string => stringifyData(dataSent) || '';

export const getResponseBody = async (responseType: string, response?: any): Promise<string> => {
  if (!response) return '';
  const _responseBody = await (responseType !== 'blob' ? response : parseResponseBlob(response));
  return stringifyData(_responseBody || '');
};

export const parseResponseBlob = async (response: Blob) => {
  const blobReader = new BlobFileReader();
  blobReader.readAsText(response);

  return await new Promise<string>((resolve, reject) => {
    const handleError = () => reject(blobReader.error);

    blobReader.addEventListener('load', () => {
      resolve(blobReader.result);
    });
    blobReader.addEventListener('error', handleError);
    blobReader.addEventListener('abort', handleError);
  });
};

export class RNRequest implements ILog {
  _id: number = -1;
  type: LogType = 'RNR';
  startTime: number = Date.now();
  readyState: number = 0;
  url: string = '';
  shortUrl?: string = '';
  method: string = '';
  status: number = -1;
  endTime: number = 0;
  timeout?: number;
  dataSent: string = '';
  requestHeaders?: any;
  responseHeaders?: any;
  responseContentType?: string;
  responseSize?: number;
  responseType?: string;
  responseURL?: string;
  response: string = '';

  constructor(attributes?: any) {
    Object.assign(this, attributes);
  }
}

export default RNRequest;
