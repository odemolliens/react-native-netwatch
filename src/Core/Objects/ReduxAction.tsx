import { ILog, LogType } from '../../types';

interface IAction {
  type: string;
  payload: any;
}

export class ReduxAction implements ILog {
  _id: number = -1;
  type: LogType = 'REDUX';
  startTime: number = Date.now();
  stringifiedAction: string = '';
  extra: string = '';
  action: IAction = { type: '__ERROR:UNDEFINED__', payload: '' };

  constructor(attributes?: any) {
    Object.assign(this, attributes);
  }
}

export default ReduxAction;
