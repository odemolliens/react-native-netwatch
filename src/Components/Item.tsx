import * as React from 'react';
import { StyleSheet, View, TouchableOpacity } from 'react-native';
import { RNRequest } from '../Core/Objects/RNRequest';
import { NRequest } from '../Core/Objects/NRequest';
import { ReduxAction } from '../Core/Objects/ReduxAction';
import { Status } from './Status';
import { getTime } from '../Utils/helpers';
import url from 'url';
import { Text, TextSecondaryColor } from '../Components/Text';
import { useContext } from 'react';
import { ThemeContext } from '../Theme';
import { ILog } from '../types';

export const ITEM_HEIGHT = 60;
export interface IProps {
  testID?: string;
  item: ILog;
  onPress: () => void;
  color: string;
}

export const Item: React.FC<IProps> = (props: IProps) => {
  const theme = useContext(ThemeContext);
  let _line1: string = '';
  let _line2: string = '';

  if (props.item instanceof ReduxAction) {
    _line1 = 'redux action';
    _line2 = props.item.stringifiedAction;
  } else {
    if (
      (props.item instanceof RNRequest || props.item instanceof NRequest) &&
      (props.item.url || props.item.shortUrl)
    ) {
      const urlObject = props.item.shortUrl ? url.parse(props.item.shortUrl) : url.parse(props.item.url);
      _line1 = (urlObject.host !== null && urlObject?.host) || '';
      _line2 = (urlObject.path !== null && urlObject?.path) || '';
    }
  }

  return (
    <TouchableOpacity
      onPress={() => props.onPress()}
      style={[
        styles.container,
        { backgroundColor: theme.secondaryLightColor, borderBottomColor: theme.textColorFour },
        { backgroundColor: props.color },
      ]}
      testID={`itemTouchable-${props.item._id}`}
    >
      <Status item={props.item} />
      <View style={styles.main}>
        <View style={styles.line}>
          <TextSecondaryColor numberOfLines={1}>{_line1}</TextSecondaryColor>
          <TextSecondaryColor style={[styles.time, { color: theme.textColorThree }]}>
            {getTime(props.item.startTime)}
          </TextSecondaryColor>
        </View>
        <View style={styles.line}>
          <Text numberOfLines={1}>{_line2}</Text>
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default Item;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    height: ITEM_HEIGHT,
    maxHeight: ITEM_HEIGHT,
    minHeight: ITEM_HEIGHT,
    borderBottomWidth: 0.5,
  },
  main: {
    flex: 1,
    paddingHorizontal: 16,
  },
  time: {
    fontSize: 10,
  },
  line: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
});
