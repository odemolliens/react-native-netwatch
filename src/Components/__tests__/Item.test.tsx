import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { Item, IProps } from '../Item';
import ReduxAction from '../../Core/Objects/ReduxAction';
import { RNRequest } from '../../Core/Objects/RNRequest';
import { NRequest } from '../../Core/Objects/NRequest';

describe('Item test suite', () => {
  let component: ShallowWrapper;
  let props: IProps;
  const onPress = jest.fn();

  it('should render Request properly', () => {
    givenProps(mockRequest);
    givenComponent();
    expect(component).toMatchSnapshot();
  });

  it('should render Native Request properly', () => {
    givenProps(mockNativeRequest);
    givenComponent();
    expect(component).toMatchSnapshot();
  });

  it('should render Request properly even no URL passed in the request to avoid crash', () => {
    givenProps(mockIncompletedRequest);
    givenComponent();
    expect(component).toMatchSnapshot();
  });

  it('should render Request properly even bad URL passed in the request to avoid crash', () => {
    givenProps(mockNullURLRequest);
    givenComponent();
    expect(component).toMatchSnapshot();
  });

  it('should render Action properly', () => {
    givenProps(mockAction);
    givenComponent();
    expect(component).toMatchSnapshot();
  });

  it('should render properly & press item', () => {
    givenProps(mockRequest);
    givenComponent();
    expect(component).toMatchSnapshot();

    whenPressingButton('itemTouchable-73');
    expect(onPress).toHaveBeenCalledTimes(1);
  });

  // GIVEN
  function givenComponent() {
    component = shallow(<Item {...props} />);
  }

  function givenProps(mockedElement) {
    props = {
      onPress,
      item: mockedElement,
      color: 'red',
    };
  }

  // UTILITIES

  function whenPressingButton(testId: string) {
    component.find(`[testID="${testId}"]`).simulate('press');
  }

  const mockAction: ReduxAction = new ReduxAction({
    _id: 73,
    startTime: 100,
    type: 'REDUX',
    action: { type: '__ERROR:UNDEFINED__', action: '' },
  });

  const mockRequest: RNRequest = new RNRequest({
    _id: 73,
    dataSent: 'dataSent',
    endTime: 1613477575757,
    method: 'GET',
    readyState: 4,
    response: 'response',
    responseContentType: 'application/json',
    responseSize: 0,
    responseType: 'blob',
    responseURL: 'https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9',
    startTime: 1613477574742,
    status: 200,
    timeout: 0,
    type: 'NR',
    url: 'https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9',
  });

  const mockNativeRequest: NRequest = new NRequest({
    _id: 73,
    dataSent: 'dataSent',
    endTime: 1613477575757,
    method: 'GET',
    readyState: 4,
    response: 'response',
    responseContentType: 'application/json',
    responseSize: 0,
    responseType: 'blob',
    responseURL: 'https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9',
    startTime: 1613477574742,
    status: 200,
    timeout: 0,
    type: 'RNR',
    url: 'https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9',
  });

  //@ts-ignore
  const mockIncompletedRequest: RNRequest = new RNRequest({
    _id: 73,
    dataSent: 'dataSent',
    endTime: 1613477575757,
    method: 'GET',
    readyState: 4,
    response: 'response',
    responseContentType: 'application/json',
    responseSize: 0,
    responseType: 'blob',
    responseURL: 'https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9',
    startTime: 1613477574742,
    status: 200,
    timeout: 0,
    type: 'RNR',
  });

  const mockNullURLRequest: RNRequest = new RNRequest({
    _id: 73,
    url: null,
    dataSent: 'dataSent',
    endTime: 1613477575757,
    method: 'GET',
    readyState: 4,
    response: 'response',
    responseContentType: 'application/json',
    responseSize: 0,
    responseType: 'blob',
    responseURL: 'https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9',
    startTime: 1613477574742,
    status: 200,
    timeout: 0,
    type: 'RNR',
  });
});
