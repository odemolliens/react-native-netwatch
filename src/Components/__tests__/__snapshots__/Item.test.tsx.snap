// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item test suite should render Action properly 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 60,
        "maxHeight": 60,
        "minHeight": 60,
        "paddingBottom": 0.5,
      },
      Object {
        "backgroundColor": "#374151",
        "borderBottomColor": "#6B7280",
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
  testID="itemTouchable-73"
>
  <Status
    item={
      ReduxAction {
        "_id": 73,
        "action": Object {
          "action": "",
          "type": "__ERROR:UNDEFINED__",
        },
        "startTime": 100,
        "stringifiedAction": "",
        "type": "REDUX",
      }
    }
  />
  <Component
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 16,
      }
    }
  >
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <TextSecondaryColor
        numberOfLines={1}
      >
        redux action
      </TextSecondaryColor>
      <TextSecondaryColor
        style={
          Array [
            Object {
              "fontSize": 10,
            },
            Object {
              "color": "#9CA3AF",
            },
          ]
        }
      >
        00:00:00
      </TextSecondaryColor>
    </Component>
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        numberOfLines={1}
      />
    </Component>
  </Component>
</ForwardRef>
`;

exports[`Item test suite should render Connection disconnected properly 1`] = `
<Component
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 60,
        "maxHeight": 60,
        "minHeight": 60,
        "paddingBottom": 0.5,
      },
      Object {
        "backgroundColor": "#374151",
        "borderBottomColor": "#6B7280",
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
  testID="itemStatic-81"
>
  <Component
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 16,
      }
    }
  >
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <TextSecondaryColor
        numberOfLines={1}
      >
        connection type : wifi
      </TextSecondaryColor>
      <TextSecondaryColor
        style={
          Array [
            Object {
              "fontSize": 10,
            },
            Object {
              "color": "#9CA3AF",
            },
          ]
        }
      >
        00:00:01
      </TextSecondaryColor>
    </Component>
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        numberOfLines={1}
      >
        You are disconnected
      </Text>
    </Component>
  </Component>
</Component>
`;

exports[`Item test suite should render Connection properly 1`] = `
<Component
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 60,
        "maxHeight": 60,
        "minHeight": 60,
        "paddingBottom": 0.5,
      },
      Object {
        "backgroundColor": "#374151",
        "borderBottomColor": "#6B7280",
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
  testID="itemStatic-81"
>
  <Component
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 16,
      }
    }
  >
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <TextSecondaryColor
        numberOfLines={1}
      >
        connection type : wifi
      </TextSecondaryColor>
      <TextSecondaryColor
        style={
          Array [
            Object {
              "fontSize": 10,
            },
            Object {
              "color": "#9CA3AF",
            },
          ]
        }
      >
        00:00:01
      </TextSecondaryColor>
    </Component>
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        numberOfLines={1}
      >
        You are connected
      </Text>
    </Component>
  </Component>
</Component>
`;

exports[`Item test suite should render Native Request properly 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 60,
        "maxHeight": 60,
        "minHeight": 60,
        "paddingBottom": 0.5,
      },
      Object {
        "backgroundColor": "#374151",
        "borderBottomColor": "#6B7280",
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
  testID="itemTouchable-73"
>
  <Status
    item={
      NRequest {
        "_id": 73,
        "dataSent": "dataSent",
        "endTime": 1613477575757,
        "method": "GET",
        "readyState": 4,
        "response": "response",
        "responseContentType": "application/json",
        "responseSize": 0,
        "responseType": "blob",
        "responseURL": "https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9",
        "shortUrl": "",
        "startTime": 1613477574742,
        "status": 200,
        "timeout": 0,
        "type": "RNR",
        "url": "https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9",
      }
    }
  />
  <Component
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 16,
      }
    }
  >
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <TextSecondaryColor
        numberOfLines={1}
      >
        run.mocky.io
      </TextSecondaryColor>
      <TextSecondaryColor
        style={
          Array [
            Object {
              "fontSize": 10,
            },
            Object {
              "color": "#9CA3AF",
            },
          ]
        }
      >
        12:12:54
      </TextSecondaryColor>
    </Component>
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        numberOfLines={1}
      >
        /v3/1a2d092a-42b2-4a89-a44f-267935dc13e9
      </Text>
    </Component>
  </Component>
</ForwardRef>
`;

exports[`Item test suite should render Request properly 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 60,
        "maxHeight": 60,
        "minHeight": 60,
        "paddingBottom": 0.5,
      },
      Object {
        "backgroundColor": "#374151",
        "borderBottomColor": "#6B7280",
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
  testID="itemTouchable-73"
>
  <Status
    item={
      RNRequest {
        "_id": 73,
        "dataSent": "dataSent",
        "endTime": 1613477575757,
        "method": "GET",
        "readyState": 4,
        "response": "response",
        "responseContentType": "application/json",
        "responseSize": 0,
        "responseType": "blob",
        "responseURL": "https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9",
        "shortUrl": "",
        "startTime": 1613477574742,
        "status": 200,
        "timeout": 0,
        "type": "NR",
        "url": "https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9",
      }
    }
  />
  <Component
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 16,
      }
    }
  >
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <TextSecondaryColor
        numberOfLines={1}
      >
        run.mocky.io
      </TextSecondaryColor>
      <TextSecondaryColor
        style={
          Array [
            Object {
              "fontSize": 10,
            },
            Object {
              "color": "#9CA3AF",
            },
          ]
        }
      >
        12:12:54
      </TextSecondaryColor>
    </Component>
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        numberOfLines={1}
      >
        /v3/1a2d092a-42b2-4a89-a44f-267935dc13e9
      </Text>
    </Component>
  </Component>
</ForwardRef>
`;

exports[`Item test suite should render Request properly even bad URL passed in the request to avoid crash 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 60,
        "maxHeight": 60,
        "minHeight": 60,
        "paddingBottom": 0.5,
      },
      Object {
        "backgroundColor": "#374151",
        "borderBottomColor": "#6B7280",
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
  testID="itemTouchable-73"
>
  <Status
    item={
      RNRequest {
        "_id": 73,
        "dataSent": "dataSent",
        "endTime": 1613477575757,
        "method": "GET",
        "readyState": 4,
        "response": "response",
        "responseContentType": "application/json",
        "responseSize": 0,
        "responseType": "blob",
        "responseURL": "https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9",
        "shortUrl": "",
        "startTime": 1613477574742,
        "status": 200,
        "timeout": 0,
        "type": "RNR",
        "url": null,
      }
    }
  />
  <Component
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 16,
      }
    }
  >
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <TextSecondaryColor
        numberOfLines={1}
      />
      <TextSecondaryColor
        style={
          Array [
            Object {
              "fontSize": 10,
            },
            Object {
              "color": "#9CA3AF",
            },
          ]
        }
      >
        12:12:54
      </TextSecondaryColor>
    </Component>
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        numberOfLines={1}
      />
    </Component>
  </Component>
</ForwardRef>
`;

exports[`Item test suite should render Request properly even no URL passed in the request to avoid crash 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 60,
        "maxHeight": 60,
        "minHeight": 60,
        "paddingBottom": 0.5,
      },
      Object {
        "backgroundColor": "#374151",
        "borderBottomColor": "#6B7280",
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
  testID="itemTouchable-73"
>
  <Status
    item={
      RNRequest {
        "_id": 73,
        "dataSent": "dataSent",
        "endTime": 1613477575757,
        "method": "GET",
        "readyState": 4,
        "response": "response",
        "responseContentType": "application/json",
        "responseSize": 0,
        "responseType": "blob",
        "responseURL": "https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9",
        "shortUrl": "",
        "startTime": 1613477574742,
        "status": 200,
        "timeout": 0,
        "type": "RNR",
        "url": "",
      }
    }
  />
  <Component
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 16,
      }
    }
  >
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <TextSecondaryColor
        numberOfLines={1}
      />
      <TextSecondaryColor
        style={
          Array [
            Object {
              "fontSize": 10,
            },
            Object {
              "color": "#9CA3AF",
            },
          ]
        }
      >
        12:12:54
      </TextSecondaryColor>
    </Component>
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        numberOfLines={1}
      />
    </Component>
  </Component>
</ForwardRef>
`;

exports[`Item test suite should render properly & press item 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 60,
        "maxHeight": 60,
        "minHeight": 60,
        "paddingBottom": 0.5,
      },
      Object {
        "backgroundColor": "#374151",
        "borderBottomColor": "#6B7280",
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
  testID="itemTouchable-73"
>
  <Status
    item={
      RNRequest {
        "_id": 73,
        "dataSent": "dataSent",
        "endTime": 1613477575757,
        "method": "GET",
        "readyState": 4,
        "response": "response",
        "responseContentType": "application/json",
        "responseSize": 0,
        "responseType": "blob",
        "responseURL": "https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9",
        "shortUrl": "",
        "startTime": 1613477574742,
        "status": 200,
        "timeout": 0,
        "type": "NR",
        "url": "https://run.mocky.io/v3/1a2d092a-42b2-4a89-a44f-267935dc13e9",
      }
    }
  />
  <Component
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 16,
      }
    }
  >
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <TextSecondaryColor
        numberOfLines={1}
      >
        run.mocky.io
      </TextSecondaryColor>
      <TextSecondaryColor
        style={
          Array [
            Object {
              "fontSize": 10,
            },
            Object {
              "color": "#9CA3AF",
            },
          ]
        }
      >
        12:12:54
      </TextSecondaryColor>
    </Component>
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        numberOfLines={1}
      >
        /v3/1a2d092a-42b2-4a89-a44f-267935dc13e9
      </Text>
    </Component>
  </Component>
</ForwardRef>
`;
